charts/opentelemetry-operator/templates/deployment.yaml

```diff
@@ -74,8 +74,10 @@ spec:
             {{- if and .Values.manager.autoInstrumentationImage.apacheHttpd.repository .Values.manager.autoInstrumentationImage.apacheHttpd.tag }}
             - --auto-instrumentation-apache-httpd-image={{ .Values.manager.autoInstrumentationImage.apacheHttpd.repository }}:{{ .Values.manager.autoInstrumentationImage.apacheHttpd.tag }}
             {{- end }}
-            {{- if .Values.manager.featureGates }}
-            - --feature-gates={{ .Values.manager.featureGates }}
+            {{- if and .Values.manager.featureGatesMap }}
+            - {{ include "opentelemetry-operator.featureGatesMap" . }}
```

Comment:
Including the `-` here will cause problems if someone adds multiple values to the map.  I'd like to see all this logic moved into a help that handles whether to use the `featureGatesMap` or `featureGates`.

Reply:
In writing the helper, I approached the problem by generating the string that goes into the `--feature-gates` and kept the `--feature-gates=` in the deployment template.